Jenkinsfile

```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env groovy
 
 common {
+    dockerPush = true
+    dockerRepos = ['confluentinc/cp-base']
+    nodeLabel = 'docker-oraclejdk7'
     slackChannel = '#tools-eng'
-    upstreamProjects = ['confluentinc/common']
+    upstreamProjects = ['confluentinc/confluent-docker-utils', 'confluentinc/common']
```

Comment:
With common building the cp-base, it guarantees the cp-base image will be available before other projects build. Now it's a leaf project, that isn't guaranteed (build could fail, but other projects will be triggered by common to rebuild even though there is no cp-base). Is that right, or are we going to change downstream projects to depend on common-docker again? I guess all downstream projects will need to switch to common-docker in order to get the utilities & build docker images?

Reply:
Answered in the other PR, but this will be transparent downstream, since almost nothing depends on these utility belt classes directly. Common keeps the Dockerfile support. Docker builds in CI use the last successful build as a base.

Reply:
Re: being a leaf node now, we can have downstream builds triggered where it matters. Right now, only some internal cloud stuff is using this new base.

Reply:
Being a leaf node / without being able to trigger downstream images to be re-built would be a problem. 

This is going to be the base for all new images, and images are meant to be built on commit via CI. That means any change to base won't cause downstream images to be rebuilt and so downstream images would be outdated during packaging time. We have many repos and some with images, so it's not easy to figure which one and having to manually trigger them is going to be error-prone. 

In terms of automation/ease of getting images for all repos, that feels like a step backward. Is there a way to make it better / at least the same as today?

Reply:
I don't follow. If it needs to trigger downstream builds, it can trigger downstream builds. Sounds like you're describing a shortcoming with how upstream projects in Jenkinsfiles are defined, not really my concern here.

Reply:
Well, I was thinking in terms of the proposal that you have where we automatically keep track of upstream/downstream dependencies based on what's declared in the POM file. Being a leaf node, downstream builds won't trigger.

In terms of today's setup using Jenkinsfile, yes, it can work that way -- and it's a good intermediate step until we get to ur proposed design. Are you saying that you will be replacing all upstreamProjects with `common` to `common-docker`? From your first comment, I wasn't sure what your plan is for that.